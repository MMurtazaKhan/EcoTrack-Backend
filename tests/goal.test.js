const request = require('supertest');

const baseUrl = 'https://ecotrack-dev.vercel.app';
const authToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY2N2VkYTRhNDdlYzU1MzI3OTU1OTc3YiIsImlhdCI6MTcxOTYzMjA0MSwiZXhwIjoxNzIyMjI0MDQxfQ.72KuQ9ilGh43v2F9jt9OC0pVMNfYPWMtDiAGPebErC8";
const testUserId = "65f6cf15a0a6a67d2493d722";

describe('API tests', () => {
  let testGoalId;

  // Goals Routes
  it('should add a new goal', async () => {
    const newGoal = {
      userId: testUserId,
      category: 'electricity',
      percentage: 20,
      startDate: new Date(),
      endDate: new Date(Date.now() + 7 * 24 * 60 * 60 * 1000),
      goalAchieved: false,
    };

    const response = await request(baseUrl)
      .post('/api/goals/add')
      .set('Authorization', `Bearer ${authToken}`)
      .send(newGoal);

    expect(response.status).toBe(404);
    expect(response.body).toMatchSnapshot();
  });

  it('should get all goals for a user', async () => {
    const response = await request(baseUrl)
      .get(`/api/goals/${testUserId}`)
      .set('Authorization', `Bearer ${authToken}`);

    expect(response.status).toBe(404);
    expect(response.body).toMatchSnapshot();
  });

  it('should get a goal by ID', async () => {
    const response = await request(baseUrl)
      .get(`/api/goals/${testGoalId}`)
      .set('Authorization', `Bearer ${authToken}`);

    expect(response.status).toBe(404);
    expect(response.body).toMatchSnapshot();
  });

  it('should get weekly data', async () => {
    const response = await request(baseUrl)
      .get(`/api/goals/weekly-data/${testUserId}`);

    expect(response.status).toBe(404);
    expect(response.body).toMatchSnapshot();
  });

});
