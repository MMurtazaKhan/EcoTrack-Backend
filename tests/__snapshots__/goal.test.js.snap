// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API tests should add a new goal 1`] = `
{
  "message": "Not Found: /api/goals/add",
  "stack": "Error: Not Found: /api/goals/add
    at notFound (file:///var/task/backend/middleware/error.js:6:17)
    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/var/task/node_modules/express/lib/router/index.js:328:13)
    at /var/task/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/var/task/node_modules/express/lib/router/index.js:346:12)
    at next (/var/task/node_modules/express/lib/router/index.js:280:10)
    at cors (/var/task/node_modules/cors/lib/index.js:188:7)
    at /var/task/node_modules/cors/lib/index.js:224:17
    at originCallback (/var/task/node_modules/cors/lib/index.js:214:15)
    at /var/task/node_modules/cors/lib/index.js:219:13",
}
`;

exports[`API tests should get a goal by ID 1`] = `
{
  "message": "Not Found: /api/goals/undefined",
  "stack": "Error: Not Found: /api/goals/undefined
    at notFound (file:///var/task/backend/middleware/error.js:6:17)
    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/var/task/node_modules/express/lib/router/index.js:328:13)
    at /var/task/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/var/task/node_modules/express/lib/router/index.js:346:12)
    at next (/var/task/node_modules/express/lib/router/index.js:280:10)
    at cors (/var/task/node_modules/cors/lib/index.js:188:7)
    at /var/task/node_modules/cors/lib/index.js:224:17
    at originCallback (/var/task/node_modules/cors/lib/index.js:214:15)
    at /var/task/node_modules/cors/lib/index.js:219:13",
}
`;

exports[`API tests should get all goals for a user 1`] = `
{
  "message": "Not Found: /api/goals/65f6cf15a0a6a67d2493d722",
  "stack": "Error: Not Found: /api/goals/65f6cf15a0a6a67d2493d722
    at notFound (file:///var/task/backend/middleware/error.js:6:17)
    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/var/task/node_modules/express/lib/router/index.js:328:13)
    at /var/task/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/var/task/node_modules/express/lib/router/index.js:346:12)
    at next (/var/task/node_modules/express/lib/router/index.js:280:10)
    at cors (/var/task/node_modules/cors/lib/index.js:188:7)
    at /var/task/node_modules/cors/lib/index.js:224:17
    at originCallback (/var/task/node_modules/cors/lib/index.js:214:15)
    at /var/task/node_modules/cors/lib/index.js:219:13",
}
`;

exports[`API tests should get weekly data 1`] = `
{
  "message": "Not Found: /api/goals/weekly-data/65f6cf15a0a6a67d2493d722",
  "stack": "Error: Not Found: /api/goals/weekly-data/65f6cf15a0a6a67d2493d722
    at notFound (file:///var/task/backend/middleware/error.js:6:17)
    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/var/task/node_modules/express/lib/router/index.js:328:13)
    at /var/task/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/var/task/node_modules/express/lib/router/index.js:346:12)
    at next (/var/task/node_modules/express/lib/router/index.js:280:10)
    at cors (/var/task/node_modules/cors/lib/index.js:188:7)
    at /var/task/node_modules/cors/lib/index.js:224:17
    at originCallback (/var/task/node_modules/cors/lib/index.js:214:15)
    at /var/task/node_modules/cors/lib/index.js:219:13",
}
`;
